### ISSUE TITLE
MooseMeshUtils::getSubdomainIDs does not seem throwing errors when a non-existing subdomain name is passed

### ISSUE THREAD

**Issue by YaqiWang** [idaholab/moose#20147]
## Reason
When calling this function, we expect to get valid subdomain IDs with subdomain names for a mesh. If the subdomain names passed in are invalid, we should throw an error. This is desired by `SubdomainExtraElementIDGenerator` mesh generator, which calls this function. Because we currently do not have this error throwing, the code does not error out properly when users provide invalid subdomain names.

## Design
Add the code in the function in `MooseMeshUtils`. Other functions in there could also require similar improvement.

## Impact
Detect input error earlier when users provide wrong subdomain names in the input.

**Comment by loganharbour** [comment]
I disagree. You should be checking for it in your object so that you can provide more context to the error (via `MooseObject:{moose/param}Error`). This is intentional.

**Comment by YaqiWang** [comment]
That is a good argument, but is MooseMeshUtils supposed to provide functionality like this shared by all objects? Maybe libMesh has already a function to check if a list of subdomain names are contained in the mesh?

**Comment by YaqiWang** [comment]
I still do not think this is not proper designed. The interface of the function is suggesting getting the subdomain ids for a mesh with names, but the function can return seemingly valid ids which are not present in the mesh. I will feel better if the interface changes to something like
```
bool getSubdomainIDs(const libMesh::MeshBase & mesh,
                                        const std::vector<SubdomainName> & subdomain_name,
                                        std::vector<subdomain_id_type> & subdomain_ids);
```

**Comment by loganharbour** [comment]
@YaqiWang and I came to a general agreement that adding a method like this:

```c++
std::set<subdomain_id_type>
missingSubdomainIDs(const libMesh::MeshBase & mesh, const std::vector<subdomain_id_type> & ids)
{
  std::set<subdomain_id_type> mesh_ids;
  mesh.subdomain_ids(mesh_ids);

  std::set<subdomain_id_type> missing_ids;
  for (const auto id : ids)
    if (!mesh_ids.count(id))
      missing_ids.insert(id);
  return missing_ids;
}
```

to `MooseMeshUtils` is the best thing to do here as to not interrupt the current behavior of `getSubdomainIDs()`. We can use this in generators/etc to validate IDs.
